search.py

```diff
@@ -0,0 +1,36 @@
+def search(pattern, url, extension, local_boolean, local_dir):
+    from urllib import request
+    import configparser
+    import shutil
+    import json
+    import os
+    import re
+
+    #Read config.ini
+    config = configparser.ConfigParser()
+    config.read('config.ini')
+    api_url = config['Search']['packages_search_remote']
+    files = []
+
+    #Get search.json
+    with request.urlopen(api_url) as response, open('search.json', 'wb') as out_file:
+        shutil.copyfileobj(response, out_file)
+
+    #Parse search.json
+    with open('search.json', "r") as search_file:
+        data = json.load(search_file)
+        for file in data:
+            name = file['name']
+            if name.endswith(extension):
+                files.append(name.replace('.sh', ''))
+
+    #Get local files
+    if local_boolean:
+        for file in os.listdir(local_dir):
+            if file.endswith(extension):
+                if file.replace('.sh', '') not in files:
+                    files.append(file.replace('.sh', ''))
+    #Search
+    r = re.compile(pattern)
+    result = list(filter(r.match, files))
```

Comment:
I dont understand the match method usage

Reply:
re.compile makes the regex pattern from the input. The r.match takes that specific pattern and goes through the files list with filter and makes a new list that is assigned to result. This list is the result of the regex search.